### PR TITLE
increase max doc length limit: 1M -> 4M

### LINKED ISSUE #817: Document limits: increase max doc length limit: 1M -> 4M
As part of evaluation of Document limits (see `DocumentLimitsConfig.java` / `CONFIGURATION.md`), we want to relax the default limit for the maximum Document size from 1 million to 4 million characters.

### PR DESCRIPTION
**Pull request by Yuqi-Du**
document max-size 1m characters -> 4m characters 

**Which issue(s) this PR fixes**:
Fixes #817 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java [hunk 1]
```diff
@@ -237,15 +238,14 @@ private void traverseVector(JsonPath path, JsonNode value, ShredListener callbac
   }
 
   private void validateDocumentSize(DocumentLimitsConfig limits, String docJson) {
+    Log.error("! " + docJson.length());
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java:241]: These were probably left by accident, remove :)

File: src/test/java/io/stargate/sgv2/jsonapi/service/shredding/ShredderDocLimitsTest.java [hunk 2]
```diff
@@ -37,16 +37,16 @@ public class ShredderDocLimitsTest {
   class ValidationDocSizeViolations {
     @Test
     public void allowBigButNotTooBigDoc() {
-      // Given we fail at 1 meg, let's try 800k (8 x 10 x 10k)
-      final ObjectNode bigDoc = createBigDoc(8, 10);
+      // Given we fail at 4 meg, let's try 600k (8 x 5 x 7.5k)
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/service/shredding/ShredderDocLimitsTest.java:40]: Would be nice to try over 1 meg document to see limit is increased?

**Review by tatu-at-datastax — CHANGES_REQUESTED** [review]
Need to remove log statements, otherwise looks good.

**Review by tatu-at-datastax — APPROVED** [review]

**Review by tatu-at-datastax — APPROVED** [review]

**Review by vkarpov15 — COMMENTED** [review]
LGTM :+1:

**Pull request merged**